File: 412-fizz-buzz/main.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        
        def is_divisible_by(v: int, div: int) -> bool:
            if v % div != 0:
                return False
                
            return True

        ans = []
        for i in range(1, n + 1):
            if is_divisible_by(i, 15):
                ans.append("FizzBuzz")
            elif is_divisible_by(i, 3):
                ans.append("Fizz")
            elif is_divisible_by(i, 5):
                ans.append("Buzz")
            else:
                ans.append(str(i))

        return ans

File: 412-fizz-buzz/main.go
func fizzBuzz(n int) []string {
    
    var ans []string
    for i := 1; i <= n; i++ {
        if isDivisibleBy(i, 15) {
            ans = append(ans, "FizzBuzz")
        } else if isDivisibleBy(i, 3) {
            ans = append(ans, "Fizz")
        } else if isDivisibleBy(i, 5) {
            ans = append(ans, "Buzz")
        } else {
            ans = append(ans, strconv.Itoa(i))
        }
    }
    return ans

}


func isDivisibleBy(v, div int) bool {
    return v % div == 0 
}
